src/main/java/com/management/HumanResources/model/TimeOff.java

```diff
@@ -12,12 +13,61 @@
     private boolean approved;
     private boolean reviewed;
 
+    /**
+     * Returns the zero-based time off start day of the week where zero is Monday.
+     */
+    public int getStartDayOfWeek() {
+        return start.getDayOfWeek().getValue() - 1; // -1 because Mon = 1 and is the first day of the week.
+    }
+
+    /**
+     * Returns the zero-based time off end day of the week where zero is Monday.
+     */
+    public int getEndDayOfWeek() {
+        return end.getDayOfWeek().getValue() - 1; // -1 because Mon = 1 and is the first day of the week.
+    }
+
     public boolean isSameDay() {
         Instant startDay = start.toInstant(ZoneOffset.UTC).truncatedTo(ChronoUnit.DAYS);
         Instant endDay = end.toInstant(ZoneOffset.UTC).truncatedTo(ChronoUnit.DAYS);
         return startDay.equals(endDay);
     }
 
+    /**
+     * Return true if the time off start and end are in the correct order and on the same week.
+     */
+	public boolean isValid() {
+        return isStartBeforeEnd() && isSameWeek();
+    }
+
+    public boolean isStartBeforeEnd() {
+        return start.compareTo(end) <= 0;
+    }
+
+    public boolean isSameWeek() {
+        final TemporalField weekOfYear = WeekFields.of(Locale.getDefault()).weekOfWeekBasedYear();
+        return start.get(weekOfYear) * start.getYear() == end.get(weekOfYear) * end.getYear()
+            || (end.getDayOfWeek().equals(DayOfWeek.MONDAY) && end.getHour() == 0 && end.getMinute() == 0);
```

Comment:
I'm going to change the whole method because it is actually all wrong now that I think about it. For example, the first week of 2021 starts on Monday 12/28/2020 and ends on 1/3/2021, but 1*2020 is not equal to 1*2021